### PR TITLE
add types for lzma

### PR DESCRIPTION
**Pull request by janKir**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by pikadun — APPROVED** [review]

File: types/lzma/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Type definitions for lzma 2.3
+// Project: http://nmrugg.github.com/LZMA-JS/
```
[the remark this answers is not in the document]
Comment (janKir) [reply types/lzma/index.d.ts:2]: You're right, I will fix this.
Although this should be a separate issue, note that I used `dts-gen` as described in the README to generate this. Also the generated `tsconfig.json` was missing the `strictFunctionTypes` option.

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## lzma ([<kbd>unpkg</kbd>](https://unpkg.com/browse/lzma@latest/))
was missing the following properties:
1. LZMA
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 63a059beab7678a0fcf8b0c12aff549deb300ef8
</p>

File: types/lzma/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Type definitions for lzma 2.3
+// Project: https://github.com/LZMA-JS/LZMA-JS
+// Definitions by: janKir <https://github.com/janKir>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ If this module has methods, declare them as functions like so.
+ */
+export function compress(
+    input: string | Uint8Array,
```
Comment (demensky) [inline types/lzma/index.d.ts:9]: https://github.com/LZMA-JS/LZMA-JS#how-to-use

> A Node.js Buffer or a Uint8Array instance counts as an array of bytes.

Does the definition fully cover supported types?

**Review by demensky — COMMENTED** [review]
What do you think about not using `any` in handlers? ([Common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) → [Return Types of Callbacks](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#return-types-of-callbacks))

**Pull request closed without merging**